### PR TITLE
Upgraded terraform and terraform oci provider to 0.12.X version

### PR DESCRIPTION
**Pull request by marinakog**
https://build.weblogick8s.org:8443/job/wko-oke-nightly-seqential/258/

### CODE REVIEW

File: integration-tests/src/test/resources/oke/terraform/cluster.tf [hunk 1]
```diff
@@ -2,62 +2,103 @@
...
-variable "node_pool_initial_node_labels_key" { default = "key" }
-variable "node_pool_initial_node_labels_value" { default = "value" }
-variable "node_pool_kubernetes_version" { default = "v1.11.5" }
-variable "node_pool_name" { default = "tfTestCluster_workers" }
-variable "node_pool_node_image_name" { default = "Oracle-Linux-7.6" }
-variable "node_pool_node_shape" { default = "VM.Standard2.1" }
-variable "node_pool_quantity_per_subnet" { default = 2 }
-variable "node_pool_ssh_public_key" { }
+variable "cluster_kubernetes_version" {
+  default = "v1.15.7"
```
Comment (rjeberhard) [inline integration-tests/src/test/resources/oke/terraform/cluster.tf:6]: This default seems off... The default OKE K8s version is 1.19.7 and the minimum is 1.17.9.

File: integration-tests/src/test/resources/oke/terraform/cluster.tf [hunk 2]
```diff
@@ -2,62 +2,103 @@
...
+variable "node_pool_initial_node_labels_key" {
+  default = "key"
+}
+
+variable "node_pool_initial_node_labels_value" {
+  default = "value"
+}
+
+variable "node_pool_kubernetes_version" {
+  default = "v1.11.5"
```
Comment (rjeberhard) [inline integration-tests/src/test/resources/oke/terraform/cluster.tf:38]: Similarly, this seems way too low.

File: integration-tests/src/test/resources/oke/terraform/kube_config.tf [hunk 3]
```diff
@@ -3,15 +3,21 @@
...
 */
 
-variable "cluster_kube_config_expiration" { default = 2592000 }
-variable "cluster_kube_config_token_version" { default = "1.0.0" }
+variable "cluster_kube_config_expiration" {
+  default = 2592000
+}
+
+variable "cluster_kube_config_token_version" {
+  default = "1.0.0"
```
Comment (rjeberhard) [inline integration-tests/src/test/resources/oke/terraform/kube_config.tf:11]: Is the 1.0 token still supported. They are on 2.0, correct?

File: integration-tests/src/test/resources/oke/terraform/cluster.tf (hunk 1 above)
Comment (marinakog) [reply integration-tests/src/test/resources/oke/terraform/cluster.tf:6]: it is overwritten by provided variable in the Tovar file, but I will change it minimum

File: integration-tests/src/test/resources/oke/terraform/cluster.tf (hunk 2 above)
Comment (marinakog) [reply integration-tests/src/test/resources/oke/terraform/cluster.tf]: done

File: integration-tests/src/test/resources/oke/terraform/kube_config.tf (hunk 3 above)
Comment (marinakog) [reply integration-tests/src/test/resources/oke/terraform/kube_config.tf]: switched to 2.0

File: integration-tests/src/test/resources/oke/terraform/cluster.tf (hunk 1 above)
Comment (marinakog) [reply integration-tests/src/test/resources/oke/terraform/cluster.tf]: done

**Comment by marinakog** [comment]
latest run https://build.weblogick8s.org:8443/job/wko-oke-nightly-seqential/260/

File: integration-tests/src/test/resources/oke/terraform/cluster.tf [hunk 4]
```diff
@@ -2,62 +2,103 @@
...
-        subnet_ids = ["${oci_core_subnet.oke-subnet-worker-1.id}", "${oci_core_subnet.oke-subnet-worker-2.id}"]
+  #Required
+  cluster_id         = oci_containerengine_cluster.tfsample_cluster.id
+  compartment_id     = var.compartment_ocid
+  kubernetes_version = var.node_pool_kubernetes_version
+  name               = var.node_pool_name
+  node_image_name    = var.node_pool_node_image_name
+  node_shape         = var.node_pool_node_shape
+
+  #subnet_ids = ["${oci_core_subnet.oke-subnet-worker-1.id}", "${oci_core_subnet.oke-subnet-worker-2.id}","${oci_core_subnet.oke-subnet-worker-3.id}"]
```
Comment (xiancao) [inline integration-tests/src/test/resources/oke/terraform/cluster.tf:93]: delete commented out line

**Review by anpanigr — COMMENTED** [review]
LGTM

File: integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf [hunk 5]
```diff
@@ -2,104 +2,156 @@
...
-        subnet_ids = ["${oci_core_subnet.oke-subnet-worker-1.id}", "${oci_core_subnet.oke-subnet-worker-2.id}"]
+  #Required
+  cluster_id         = oci_containerengine_cluster.tfsample_cluster1.id
+  compartment_id     = var.compartment_ocid
+  kubernetes_version = var.node_pool_kubernetes_version
+  name               = var.node_pool_name1
+  node_image_name    = var.node_pool_node_image_name
+  node_shape         = var.node_pool_node_shape
+
+  #subnet_ids = ["${oci_core_subnet.oke-subnet-worker-1.id}", "${oci_core_subnet.oke-subnet-worker-2.id}","${oci_core_subnet.oke-subnet-worker-3.id}"]
```
Comment (xiancao) [inline integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf:125]: delete commented out line

File: integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf [hunk 6]
```diff
@@ -2,104 +2,156 @@
...
-  node_image_name = "${var.node_pool_node_image_name}"
-  node_shape = "${var.node_pool_node_shape}"
+  cluster_id         = oci_containerengine_cluster.tfsample_cluster2.id
+  compartment_id     = var.compartment_ocid
+  kubernetes_version = var.node_pool_kubernetes_version
+  name               = var.node_pool_name2
+  node_image_name    = var.node_pool_node_image_name
+  node_shape         = var.node_pool_node_shape
+
   #subnet_ids = ["${oci_core_subnet.oke-subnet-worker-1.id}", "${oci_core_subnet.oke-subnet-worker-2.id}","${oci_core_subnet.oke-subnet-worker-3.id}"]
```
Comment (xiancao) [inline integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf:142]: delete commented out lines

**Review by xiancao — APPROVED** [review]
LG, some minor comments.

File: integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf (hunk 6 above)
Comment (marinakog) [reply integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf:142]: I left them intentionally , so we can switch between2 and 3 nodes in cluster

File: integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf (hunk 5 above)
Comment (marinakog) [reply integration-tests/src/test/resources/oke/terraform/twoclusters/cluster.tf:125]: I left them intentionally , so we can switch between2 and 3 nodes in cluster

File: integration-tests/src/test/resources/oke/terraform/cluster.tf (hunk 4 above)
Comment (marinakog) [reply integration-tests/src/test/resources/oke/terraform/cluster.tf:93]: I left them intentionally , so we can switch between2 and 3 nodes in cluster

**Review by anpanigr — APPROVED** [review]

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**